src/ring_theory/etale.lean

```diff
@@ -355,6 +374,36 @@ end
 
 end of_surjective
 
+section unramified_derivation
+
+open_locale tensor_product
+
+variables {R S : Type u} [comm_ring R] [comm_ring S] [algebra R S]
+
+lemma formally_unramified.iff_kaehler_differential_subsingleton :
+  formally_unramified R S ↔ subsingleton Ω[S⁄R] :=
+begin
+  split,
+  { rw ← not_nontrivial_iff_subsingleton,
+    introsI h₁ h₂,
+    obtain ⟨f₁, f₂, e⟩ := (kaehler_differential.End_equiv R S).injective.nontrivial,
+    apply e,
+    ext1,
+    apply formally_unramified.lift_unique' _ _ _ _ (f₁.2.trans f₂.2.symm),
+    rw [← alg_hom.to_ring_hom_eq_coe, alg_hom.ker_ker_sqare_lift],
+    exact ⟨_, ideal.cotangent_ideal_square _⟩ },
+  { introI H,
+    constructor,
+    introsI B _ _ I hI f₁ f₂ e,
+    letI := f₁.to_ring_hom.to_algebra,
+    haveI := is_scalar_tower.of_algebra_map_eq' (f₁.comp_algebra_map).symm,
+    have := ((kaehler_differential.linear_map_equiv_derivation R S).to_equiv.trans
+      (derivation_to_square_zero_equiv_lift I hI)).surjective.subsingleton,
+    exact subtype.ext_iff.mp (@@subsingleton.elim this ⟨f₁, rfl⟩ ⟨f₂, e.symm⟩) }
+end
```

Comment:
I think the forward direction can be made into an instance. Could you please prove that first?
Something like this (untested):
```suggestion
instance formally_unramified.kaehler_differential_subsingleton [formally_unramified R S] :
  subsingleton Ω[S⁄R] :=
begin
  rw ← not_nontrivial_iff_subsingleton,
  introsI h,
  obtain ⟨f₁, f₂, e⟩ := (kaehler_differential.End_equiv R S).injective.nontrivial,
  apply e,
  ext1,
  apply formally_unramified.lift_unique' _ _ _ _ (f₁.2.trans f₂.2.symm),
  rw [← alg_hom.to_ring_hom_eq_coe, alg_hom.ker_ker_sqare_lift],
  exact ⟨_, ideal.cotangent_ideal_square _⟩,
end

lemma formally_unramified.iff_kaehler_differential_subsingleton :
  formally_unramified R S ↔ subsingleton Ω[S⁄R] :=
begin
  split,
  { introsI, apply_instance },
  { introI H,
    constructor,
    introsI B _ _ I hI f₁ f₂ e,
    letI := f₁.to_ring_hom.to_algebra,
    haveI := is_scalar_tower.of_algebra_map_eq' (f₁.comp_algebra_map).symm,
    have := ((kaehler_differential.linear_map_equiv_derivation R S).to_equiv.trans
      (derivation_to_square_zero_equiv_lift I hI)).surjective.subsingleton,
    exact subtype.ext_iff.mp (@@subsingleton.elim this ⟨f₁, rfl⟩ ⟨f₂, e.symm⟩) }
end
```